docs/restricting-api-access.mdx

```diff
@@ -0,0 +1,169 @@
+---
+title: Restricting API access
+sidebar_position: 2
+---
+
+Even though Saleor [implements complex authorization and authentication](/api-usage/authentication.mdx) mechanisms, you might want to restrict access to your Saleor Cloud instance for specific use cases.
+
+Below, we explore two ways Saleor Cloud allows you to restrict access to the API, as well as a practical example of how to use one of them.
+
+## Allowed API origins
+
+Most times, the list of the origins you expect to call your API is known in advance. By specifying the allowed origins, you can prevent unauthorized access to your API.
+
+In the project view of your Saleor Cloud dashboard, you can customize the allowed origins. The default option is to "Allow all", but you can also specify a list of origins or set it to "Dashboard only".
+
+:::info
+
+Changing the settings to only allow the selected origins can affect the installed apps. If you want your app to call the Saleor API from the client, you need to add the app's origin to the list of allowed origins. You can read more about app's access scopes in the [App permissions](developer/extending/apps/architecture/app-permissions.mdx#access-scopes) document.
+
+:::
+
+Here is how an example list of allowed origins may look:
+
+- `https://your-storefront.com` - your storefront
+- `https://your-app-1.com` - your first app (if it is calling the Saleor API from the client)
+- `https://your-app-2.com` - your second app (if it is calling the Saleor API from the client)
+
+You don't need to specify the Saleor Cloud dashboard origin, as it is always allowed to call the Saleor API.
+
+## API Password Protection
+
+Another way to restrict access to your Saleor Cloud instance is to protect it with a password through [Basic Auth](https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication#basic_authentication_scheme). This way, you can ensure that only those who know the credentials can access the API.
+
+:::important
+
+After enabling password protection, users trying to open the Saleor Dashboard will be prompted to enter the Basic Auth username and password, besides their own Saleor Cloud credentials.
+
+:::
+
+Combined with a server-side proxy, this can be a powerful tool to obscure the Saleor API from unauthorized access. We will explore this in more detail in the example below.
+
+## Example: Password-protected API proxy
+
+### Overview
+
+Let's imagine you have a storefront that communicates with Saleor API. Your storefront, naturally, implements some sort of a checkout flow. In Saleor, most checkout actions don't require the user to be authenticated. It is done so by design - you want to allow users to manage their cart without being logged in.
```

Comment:
Would rather choose to describe in detail the usecase which is implemented in the example - getting the list of products. The webhooks rant is adding unnecessary context here.

It is a valid usecase - by default no authentication is required to get the list of available products. If shop owner wants to restrict that data only for the logged in users, Password API Protection for the API is the solution.

Reply:
addressed